state.go

```diff
@@ -530,6 +574,27 @@ type FinalState struct {
 	BaseState
 }
 
-func (f *FinalState) Handle(c Controller) (State, bool) {
+func (f *FinalState) Handle(ctx *StateContext, c Controller) (State, bool) {
 	panic("reached final state")
 }
+
+func StoreStateData(store Store, sd StateData) error {
```

Comment:
TODO: as decided today we will evaluate storing this in some DB, which will be a part of another task.